File: gitdot-web/packages/api/src/endpoint/user/list-user-commits.ts
import { z } from "zod";
import { RepositoryCommitResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListUserCommitsRequest = z.object({});
export type ListUserCommitsRequest = z.infer<typeof ListUserCommitsRequest>;

export const ListUserCommitsResponse = z.array(RepositoryCommitResource);
export type ListUserCommitsResponse = z.infer<typeof ListUserCommitsResponse>;

export const ListUserCommits = {
  path: "/user/{user_name}/commits",
  method: "GET",
  request: ListUserCommitsRequest,
  response: ListUserCommitsResponse,
} as const satisfies Endpoint;
export type ListUserCommits = typeof ListUserCommits;

File: gitdot-api/src/endpoint/user/list_user_commits.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryCommitResource};

pub struct ListUserCommits;

impl Endpoint for ListUserCommits {
    const PATH: &'static str = "/user/{user_name}/commits";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListUserCommitsRequest;
    type Response = ListUserCommitsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct ListUserCommitsRequest {}

pub type ListUserCommitsResponse = Vec<RepositoryCommitResource>;
